applications/RANSApplication/custom_processes/rans_line_output_process.cpp

```diff
@@ -0,0 +1,516 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System includes
+#include <fstream>
+#include <functional>
+
+// External includes
+
+// Project includes
+#include "includes/data_communicator.h"
+#include "includes/define.h"
+#include "input_output/logger_output.h"
+#include "utilities/brute_force_point_locator.h"
+
+// Application includes
+#include "custom_utilities/rans_calculation_utilities.h"
+
+// Include base h
+#include "rans_line_output_process.h"
+
+namespace Kratos
+{
+RansLineOutputProcess::RansLineOutputProcess(
+    Model& rModel,
+    Parameters rParameters)
+: Process(), mrModel(rModel)
+{
+    KRATOS_TRY
+
+    rParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+
+    mModelPartName = rParameters["model_part_name"].GetString();
+    mWriteHeader = rParameters["write_header_information"].GetBool();
+
+    const Vector& r_start_point = rParameters["start_point"].GetVector();
+    KRATOS_ERROR_IF(r_start_point.size() != 3)
+        << "\"start_point\" should be given as 3-valued vector. [ "
+           "start_point = "
+        << r_start_point << " ]\n";
+    const Vector& r_end_point = rParameters["end_point"].GetVector();
+    KRATOS_ERROR_IF(r_end_point.size() != 3)
+        << "\"end_point\" should be given as 3-valued vector. [ "
+           "end_point = "
+        << r_end_point << " ]\n";
+    noalias(mStartPoint) = r_start_point;
+    noalias(mEndPoint) = r_end_point;
+
+    mNumberOfSamplingPoints = rParameters["number_of_sampling_points"].GetInt();
+    KRATOS_ERROR_IF(mNumberOfSamplingPoints <= 2)
+        << "Sampling points should be greater than 2. [ "
+           "number_of_sampling_points = "
+        << mNumberOfSamplingPoints << " ]\n";
+
+    mVariableNames = rParameters["variable_names_list"].GetStringArray();
+    mIsHistoricalValue = rParameters["historical_value"].GetBool();
```

Comment:
I'd make two lists, one with historical variables and one with non-historical ones. In that way once process instance could export both historical and non-historical values at the same time.

Reply:
done :), now it supports historical/non-historical double, array3, array4, array6, array9, vector and matrix by templating :)